test/Conversion/TritonArithToLinalg/reduce_extend_fp32_precision.mlir

```diff
@@ -0,0 +1,111 @@
+// RUN: triton-shared-opt --triton-to-structured --canonicalize --cse --triton-arith-to-linalg %s | FileCheck %s
+
+module {
+  tt.func public @fn1(%arg0: !tt.ptr<f16> , %arg1: !tt.ptr<f16> ) attributes {noinline = false} {
+    %0 = tt.make_range {end = 32 : i32, start = 0 : i32} : tensor<32xi32>
+    %1 = tt.splat %arg0 : !tt.ptr<f16> -> tensor<32x!tt.ptr<f16>>
+    %2 = tt.addptr %1, %0 : tensor<32x!tt.ptr<f16>>, tensor<32xi32>
+    %3 = tt.load %2 : tensor<32x!tt.ptr<f16>>
+    %4 = "tt.reduce"(%3) <{axis = 0 : i32}> ({
+    ^bb0(%arg2: f16 , %arg3: f16 ):
+      %12 = arith.addf %arg2, %arg3 : f16
+      tt.reduce.return %12 : f16
+    }) : (tensor<32xf16>) -> f16
+    %5 = arith.extf %3 : tensor<32xf16> to tensor<32xf32>
+    %6 = arith.extf %4 : f16 to f32
+    %7 = tt.splat %6 : f32 -> tensor<32xf32>
+    %8 = arith.divf %5, %7 : tensor<32xf32>
+    %9 = tt.splat %arg1 : !tt.ptr<f16> -> tensor<32x!tt.ptr<f16>>
+    %10 = tt.addptr %9, %0 : tensor<32x!tt.ptr<f16>>, tensor<32xi32>
+    %11 = arith.truncf %8 : tensor<32xf32> to tensor<32xf16>
+    tt.store %10, %11 : tensor<32x!tt.ptr<f16>>
+    tt.return
+  }
+
+  tt.func public @fn2(%arg0: !tt.ptr<f16> , %arg1: !tt.ptr<f16> ) attributes {noinline = false} {
+    %0 = tt.make_range {end = 32 : i32, start = 0 : i32} : tensor<32xi32>
+    %1 = tt.splat %arg0 : !tt.ptr<f16> -> tensor<32x!tt.ptr<f16>>
+    %2 = tt.addptr %1, %0 : tensor<32x!tt.ptr<f16>>, tensor<32xi32>
+    %3 = tt.load %2 : tensor<32x!tt.ptr<f16>>
+    %4 = "tt.reduce"(%3) <{axis = 0 : i32}> ({
+    ^bb0(%arg2: f16 , %arg3: f16 ):
+      %9 = arith.addf %arg2, %arg3 : f16
+      tt.reduce.return %9 : f16
+    }) : (tensor<32xf16>) -> f16
```

Comment:
Do we have tests for the case of bf16 return and f32 return?

Reply:
thanks! added f32 and bf16 tests in the same file